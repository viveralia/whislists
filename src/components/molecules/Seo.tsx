import { FC } from "react";
import { Helmet } from "react-helmet";

import { SITE_NAME, SITE_DESCRIPTION } from "../../config/metadata";

export interface SeoProps {
  pageTitle: string;
  pageDescription?: string;
}

const Seo: FC<SeoProps> = ({ pageTitle, pageDescription }) => {
  return (
    <Helmet>
      <title>
        {pageTitle} | {SITE_NAME}
      </title>
      <meta name="description" content={pageDescription || SITE_DESCRIPTION} />
    </Helmet>
  );
};

export default Seo;
