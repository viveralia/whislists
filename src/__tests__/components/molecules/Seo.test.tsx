import { shallow } from "enzyme";

import Seo from "../../../components/molecules/Seo";

it("should render correctly", () => {
  const wrapper = shallow(<Seo pageTitle="Test" />);
  expect(wrapper).toMatchSnapshot();
});
