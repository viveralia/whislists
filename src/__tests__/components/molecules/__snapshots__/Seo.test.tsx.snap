// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <title>
    Test
     | 
    Wishlistify
  </title>
  <meta
    content="Create wishlists and add products from all around the web. Share with your friends."
    name="description"
  />
</HelmetWrapper>
`;
